presto-hive/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -0,0 +1,354 @@
+    {
+        if (storage.isSorted() || storage.isSkewed()) {
+            throw new IllegalArgumentException("Writing to sorted and/or skewed table/partition is not supported");
+        }
+        SerDeInfo serdeInfo = new SerDeInfo();
+        serdeInfo.setName(tableName);
+        serdeInfo.setSerializationLib(storage.getStorageFormat().getSerDeNullable());
+        serdeInfo.setParameters(storage.getSerdeParameters());
+
+        StorageDescriptor sd = new StorageDescriptor();
+        sd.setLocation(emptyToNull(storage.getLocation()));
+        sd.setCols(columns.stream()
+                .map(MetastoreUtil::toMetastoreApiFieldSchema)
+                .collect(Collectors.toList()));
+        sd.setSerdeInfo(serdeInfo);
+        sd.setInputFormat(storage.getStorageFormat().getInputFormatNullable());
+        sd.setOutputFormat(storage.getStorageFormat().getOutputFormatNullable());
+        sd.setParameters(ImmutableMap.of());
+
+        Optional<HiveBucketProperty> bucketProperty = storage.getBucketProperty();
+        if (bucketProperty.isPresent()) {
+            sd.setNumBuckets(bucketProperty.get().getBucketCount());
+            sd.setBucketCols(bucketProperty.get().getBucketedBy());
+        }
+
+        return sd;
+    }
+
+    private static FieldSchema toMetastoreApiFieldSchema(Column column)
+    {
+        return new FieldSchema(column.getName(), column.getType().getHiveTypeName(), column.getComment().orElse(null));
+    }
+
+    private static Column fromMetastoreApiFieldSchema(FieldSchema fieldSchema)
+    {
+        return new Column(fieldSchema.getName(), HiveType.valueOf(fieldSchema.getType()), makeOptional(fieldSchema.getComment()));
+    }
+
+    private static void fromMetastoreApiStorageDescriptor(StorageDescriptor storageDescriptor, Storage.Builder builder, String tablePartitionName)
+    {
+        SerDeInfo serdeInfo = storageDescriptor.getSerdeInfo();
+        if (serdeInfo == null) {
+            throw new PrestoException(HIVE_INVALID_METADATA, "Table storage descriptor is missing SerDe info");
+        }
+
+        builder.setStorageFormat(StorageFormat.createNullable(serdeInfo.getSerializationLib(), storageDescriptor.getInputFormat(), storageDescriptor.getOutputFormat()))
+                .setLocation(nullToEmpty(storageDescriptor.getLocation()))
+                .setBucketProperty(HiveBucketProperty.fromStorageDescriptor(storageDescriptor, tablePartitionName))
+                .setSorted(storageDescriptor.isSetSortCols() && !storageDescriptor.getSortCols().isEmpty())
+                .setSkewed(storageDescriptor.isSetSkewedInfo() && storageDescriptor.getSkewedInfo().isSetSkewedColNames() && !storageDescriptor.getSkewedInfo().getSkewedColNames().isEmpty())
+                .setSerdeParameters(serdeInfo.getParameters() == null ? ImmutableMap.of() : serdeInfo.getParameters());
+    }
+
+    private static Optional<String> makeOptional(String string)
```

Comment:
How about we just drop this method and use this code: `Optional.ofNullable(emptyToNull(""));`  the `emptyToNull` method is in guava.